platform/linux-generic/odp_random_std.c

```diff
@@ -40,38 +41,35 @@ static int32_t _random_data(uint8_t *buf, uint32_t len, uint64_t *seed)
 {
 	const uint32_t ret = len;
 
-	if (!_ODP_UNALIGNED) {
-		uint32_t r = xorshift64s32(seed);
+	if (!_ODP_UNALIGNED && (uintptr_t)buf & 3) {
+		for (uint32_t i = 0; i < len / 4; i++) {
+			uint32_t r = xorshift64s32(seed);
 
-		if (((uintptr_t)buf & 1) && len) {
-			*(uint8_t *)buf = r & 0xff;
-			r >>= 8;
-			buf += 1;
-			len -= 1;
+			memcpy(buf, &r, 4);
+			buf += 4;
 		}
 
-		if (((uintptr_t)buf & 2) && len >= 2) {
-			*(uint16_t *)(uintptr_t)buf = r & 0xffff;
-			buf += 2;
-			len -= 2;
-		}
+		len &= 3;
 
-		for (uint32_t i = 0; i < len / 4; i++) {
-			*(uint32_t *)(uintptr_t)buf = xorshift64s32(seed);
-			buf += 4;
-		}
-	} else {
-		for (uint32_t i = 0; i < len / 4; i++) {
-			*(odp_una_u32_t *)buf = xorshift64s32(seed);
-			buf += 4;
+		if (len) {
+			uint32_t r = xorshift64s32(seed);
+
+			memcpy(buf, &r, len);
 		}
+
+		return ret;
+	}
+
+	for (uint32_t i = 0; i < len / 4; i++) {
+		*(odp_una_u32_t *)buf = xorshift64s32(seed);
```

Comment:
This could be memcpy (at least if we do not care too much about _ODP_UNALIGNED 0) to avoid alignment and aliasing problems and the need to use weird type attributes. The memcpy would get optimized away when compiling for x86_64 or A64. And same comments for the other occurences of this sort of assignments.

Reply:
I'm not in favor of replacing every unaligned integer assignment with memcpy. But in any case that's outside the scope of this PR, so I left this as-is.